File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewsStationNode : public rclcpp::Node // MODIFY NAME
{
public:
    RobotNewsStationNode() : Node("robot_news_station_cpp")
    {
        this->declare_parameter<std::string>("robot_name1", "R2D2_");
        this->declare_parameter<std::string>("robot_name2", "C3P0_");
        this->declare_parameter<std::string>("text1", "Hi Vincent, this is ");

        robot_name_ = this->get_parameter("robot_name1").as_string();
        robot_name2_ = this->get_parameter("robot_name2").as_string();
        text1_ =this->get_parameter("text1").as_string();

        Publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news_cpp", 10);
        Publisher2_ = this->create_publisher<example_interfaces::msg::String>("robot_sensor_cpp", 10);
        timer_ = create_wall_timer(std::chrono::milliseconds(500),
                                   std::bind(&RobotNewsStationNode::publishNew, this));
        timer2_ = create_wall_timer(std::chrono::milliseconds(500),
                                   std::bind(&RobotNewsStationNode::publishNew2, this));
        RCLCPP_INFO(this->get_logger(), "Robot New Station has been started.");
    }

private:
    void publishNew()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = text1_ + robot_name_ + std::string(" from the Robot New Station.");
        Publisher_->publish(msg);
    }
    void publishNew2()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = std::string("Hi Fabricio, this is ") + robot_name2_ + std::string(" from the Robot New Station.");
        Publisher2_->publish(msg);
    }

    std::string robot_name_;
    std::string robot_name2_;
    std::string text1_;

    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr Publisher_;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr Publisher2_;
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::TimerBase::SharedPtr timer2_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>(); // MODIFY NAME
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String
 
 
class RobotsNewsStationNode(Node): # MODIFY NAME
    def __init__(self):
        super().__init__("robot_news_station") # MODIFY NAME

        self.robot_name_ = "C3PO"
        self.robot_name2_ ="R2D2"
        self.publisher_ = self.create_publisher(String, "robot_news_",10)
        self.publisher2_ =self.create_publisher(String, "robot_sensor_",10)
        self.timer_ = self.create_timer(0.5, self.publish_news)
        self.timer2_ = self.create_timer (1, self.publish2_news)
        self.get_logger().info("Robot News Station has been started")

    def publish2_news(self):
        msg = String()
        msg.data = "Hello Fabricio, this is " + str(self.robot_name2_) + " from the robot news station."
        self.publisher2_.publish(msg)

    def publish_news(self):
        msg = String()
        msg.data = "Hello Vincent, this is " + str(self.robot_name_) + " from the robot news station."
        self.publisher_.publish(msg)
 
 
def main(args=None):
    rclpy.init(args=args)
    node = RobotsNewsStationNode() # MODIFY NAME
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()